agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/AIAgent.kt

```diff
@@ -430,25 +430,65 @@ public open class AIAgent(
+ *                - SingleRunMode.SEQUENTIAL: Executes simultaneous tool calls sequentially.
+ *                - SingleRunMode.PARALLEL: Executes multiple tool calls in parallel.
+ * @return An instance of AIAgentStrategy configured according to the specified single-run mode.
  */
-public fun singleRunStrategy(): AIAgentStrategy = strategy("single_run") {
-    val nodeCallLLM by nodeLLMRequest("sendInput")
-    val nodeExecuteTool by nodeExecuteTool("nodeExecuteTool")
-    val nodeSendToolResult by nodeLLMSendToolResult("nodeSendToolResult")
+public fun singleRunStrategy(runMode: SingleRunMode = SingleRunMode.SINGLE): AIAgentStrategy =
+    when (runMode) {
+        SingleRunMode.SEQUENTIAL -> singleRunWithParallelAbility(false)
+        SingleRunMode.PARALLEL   -> singleRunWithParallelAbility(true)
+        SingleRunMode.SINGLE     -> singleRunModeStrategy()
+    }
+
+private fun singleRunWithParallelAbility(parallelTools: Boolean) = strategy("single_run_sequential") {
+    val nodeCallLLM by nodeLLMRequestMultiple()
+    val nodeExecuteTool by nodeExecuteMultipleTools(parallelTools = parallelTools)
+    val nodeSendToolResult by nodeLLMSendMultipleToolResults()
+
+    edge(nodeStart forwardTo nodeCallLLM)
+    edge(nodeCallLLM forwardTo nodeExecuteTool onMultipleToolCalls { true })
+    edge(nodeCallLLM forwardTo nodeFinish
+                                        onMultipleAssistantMessages { true }
+                                        transformed { it.joinToString("\n") { message -> message.content } })
+
+    edge(nodeExecuteTool forwardTo nodeSendToolResult)
+
+    edge(nodeSendToolResult forwardTo nodeFinish
+                                                onMultipleAssistantMessages { true }
+                                                transformed { it.joinToString("\n") { message -> message.content } })
+
+    edge(nodeSendToolResult forwardTo nodeExecuteTool onMultipleToolCalls { true })
+}
+
+private fun singleRunModeStrategy() = strategy("single_run") {
+    val nodeCallLLM by nodeLLMRequest()
+    val nodeExecuteTool by nodeExecuteTool()
+    val nodeSendToolResult by nodeLLMSendToolResult()
 
     edge(nodeStart forwardTo nodeCallLLM)
     edge(nodeCallLLM forwardTo nodeExecuteTool onToolCall { true })
     edge(nodeCallLLM forwardTo nodeFinish onAssistantMessage { true })
     edge(nodeExecuteTool forwardTo nodeSendToolResult)
     edge(nodeSendToolResult forwardTo nodeFinish onAssistantMessage { true })
     edge(nodeSendToolResult forwardTo nodeExecuteTool onToolCall { true })
+}
+
+/**
+ * Enum representing the modes in which a single-run strategy for an AI agent can be executed.
+ *
+ * These modes define how tasks or operations are processed during the agent's run:
+ * - SEQUENTIAL: Multiple tool calls allowed but will be executed sequentially.
+ * - PARALLEL: Tool calls executed in parallel.
+ * - SINGLE: Multiple tool calls are not allowed.
+ */
+public enum class SingleRunMode {
+    SEQUENTIAL, PARALLEL, SINGLE
```

Comment:
Let's maybe be more verbose about tools semantics here?
`SingleRunMode` -> `ToolCalls`?

`SEQUENTIAL` -> `SEQUENTIAL_TOOLS_CALLS`
`PARALLEL ` -> `PARALLEL_TOOL_CALLS`
`SINGLE` -> `INDIVIDUAL_TOOL_CALLS`

or smth similar

WDYT ?

Reply:
I do like it, but in that case naming like `SEQUENTIAL_TOOLS_CALLS` is not necessary, as we'll use it as `ToolCalls.SEQUENTIAL`. There's no need to duplicate "tool calls" naming.